third_party/nvfuser/csrc/expr_simplifier.cpp

```diff
@@ -1112,16 +1118,7 @@ bool isPositive(Val* value, const VarInfoMap& var_info) {
     return true;
   }
   if (auto ns = dynamic_cast<NamedScalar*>(value)) {
-    // TODO: currently we are implicitly assuming tensor sizes to be non-zero.
-    // For example, if I have T0[2, 0] and I do T0->merge(0), I get T0[2*0]. And
-    // during indexing, we will generate index like: (i / 0, i % 0). Strictly
-    // speaking, this is wrong, but we are not handling this edge case. We are
-    // just hoping that we will be lucky and this will work. For expression
-    // simplification, I am not considering this edge case either, because all
-    // simplifications involving div and mod requires the divisor to be
-    // non-zero, and I don't want this edge case to block the simplification of
-    // normal cases.
-    if (ns->getParallelDim().has_value() || ns->isTensorSize()) {
```

Comment:
It doesn't make sense to assume tensor size to be positive globally. All we want is to unblock the simplification of `i / size * size + i % size -> i`, but if we do assume size to be positive, we would also simplify predicates like `0 < size` as `true`, which could potentially break zero tensor support. Instead, we should have a `preserve_error` flag for expr simplified to unblock that simplification even if we can not prove `size` to be non-zero.